src/NHibernate/SqlCommand/JoinFragment.cs

```diff
@@ -62,7 +62,7 @@ protected bool AddCondition(SqlStringBuilder buffer, SqlString on)
 		{
 			if (SqlStringHelper.IsNotEmpty(on))
 			{
-				if (!on.StartsWithCaseInsensitive(" and"))
+				if (buffer.Count > 0 && !on.StartsWithCaseInsensitive(" and"))
```

Comment:
I doubt this change is legit. Was it causing a failure not having this change done? It looks to me code expects condition clauses to always potentially starts with an " and", removing it if needed.

Reply:
The intention here is the same as in `ANSIJoinFragment.AddJoin` - avoid bare `withClause` to be added with " and" prefix to SQL. But yeah - maybe I should handle it in each dialect explicitly.

Reply:
You are right - I've removed this change. I've struggled with this " and" prefix and thought this change is required for Oracle.